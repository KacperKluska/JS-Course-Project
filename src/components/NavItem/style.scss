.navItem {
  color: white;
  cursor: pointer;
}

@media (max-width: 550px) {
  .navItem {
    padding: 0.5rem;
  }
}
