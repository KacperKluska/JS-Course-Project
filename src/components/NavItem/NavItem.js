import './style.scss';

export default function NavItem({ icon, children, onClick }) {
  return (
    <li className="navItem">
      <button type="button" onClick={onClick}>
        <i className={icon} />
        {children}
      </button>
    </li>
  );
}
