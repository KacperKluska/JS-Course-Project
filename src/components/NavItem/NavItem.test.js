import NavItem from "./NavItem";

let wrapper;

beforeAll(() => {
  wrapper = mount(<NavItem icon="icon-search">Test me</NavItem>);
});

test("renders", () => {
  expect(wrapper).not.toBeNull();
});

test("test blocks tags", () => {
  expect(wrapper.find(".navItem").type()).toBe("li");
  expect(wrapper.find(".icon-search").type()).toBe("i");
});

test("navItem value test", () => {
  expect(wrapper.find(".navItem").text()).toBe("Test me");
  expect(wrapper.find(".icon-search")).not.toBeNull();
});

test("simulate click", () => {
  const spyFunction = jest.fn();
  const newWrapper = mount(
    <NavItem icon="icon-search" onClick={spyFunction}>
      Test me
    </NavItem>
  );

  expect(spyFunction).toBeCalledTimes(0);
  newWrapper.find(".navItem").simulate("click");
  expect(spyFunction).toBeCalledTimes(1);
  newWrapper.find(".navItem").simulate("click");
  newWrapper.find(".navItem").simulate("click");
  newWrapper.find(".navItem").simulate("click");
  expect(spyFunction).toBeCalledTimes(4);
});

test("NavItem snapshot", () => {
  const container = renderer.create(
    <NavItem icon="icon-search">Test me</NavItem>
  );
  expect(container.toJSON()).toMatchSnapshot();
});
