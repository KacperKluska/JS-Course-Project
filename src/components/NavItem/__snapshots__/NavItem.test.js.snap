// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem snapshot 1`] = `
<li
  className="navItem"
>
  <i
    className="icon-search"
  />
  Test me
</li>
`;
